src/server/api/routes/users/v2/auth.js

```diff
@@ -11,26 +15,28 @@ const CLEAR_AUTH_COOKIE_OPTION = { ...AUTH_COOKIE_OPTION, maxAge: 0 };
 
 const router = Router();
 
-router.get('/v2/users/self', async (req, res, next) => {
-  try {
-    setPrivateCacheHeader(res);
-    const { user } = req.session;
-    if (user) {
+router.get(
+  '/v2/users/:wallet/self',
```

Comment:
I dont think a `/self` endpoint should has wallet as path param, just return the current session wallet in return payload..